sage-ios/SAGE/SAGE/LoginOperations.swift

```diff
@@ -0,0 +1,175 @@
+                let authToken = (KeychainWrapper.objectForKey(KeychainConstants.kAuthToken) as? String)
+                if (email != nil && authToken != nil) {
+                    LoginOperations.loginWith(email!, authToken: authToken!, completion: { (success) -> Void in
+                        if success && (KeychainWrapper.objectForKey(KeychainConstants.kUser) as! User).verified {
+                            completion(true)
+                        } else {
+                            completion(false)
+                        }
+                    })
+                } else {
+                    completion(false)
+                }
+            }
+        } else {
+            completion(false)
+        }
+    }
+    
+    static func storeUserDataInKeychain(user: User, authToken: String? = nil) {
+        if let existingUser = (KeychainWrapper.objectForKey(KeychainConstants.kUser) as? User) {
+            if User.DefaultValues.DefaultID.rawValue != user.id {
+                existingUser.id = user.id
+            }
+            if let firstName = user.firstName {
+                existingUser.firstName = firstName
+            }
+            if let lastName = user.lastName {
+                existingUser.lastName = lastName
+            }
+            if let email = user.email {
+                existingUser.email = email
+            }
+            if let school = user.school {
+                existingUser.school = school
+            }
+            if User.VolunteerLevel.Default != user.level {
+                existingUser.level = user.level
+            }
+            if User.UserRole.Default != user.role {
+                existingUser.role = user.role
+            }
+            if User.DefaultValues.DefaultHours.rawValue != user.totalHours {
+                existingUser.totalHours = user.totalHours
+            }
+            existingUser.verified = user.verified
+            KeychainWrapper.setObject(existingUser, forKey: KeychainConstants.kUser)
+        } else {
+            KeychainWrapper.setObject(user, forKey: KeychainConstants.kUser)
+            if let auth = authToken {
+                KeychainWrapper.setObject(auth, forKey: KeychainConstants.kAuthToken)
+            }
+        }
+    }
+    
+    static func createUser(firstName: String, lastName: String, email: String, password: String, school: Int, hours: Int, role: Int, photoData: String, completion: ((Bool) -> Void)) {
+        
+        let operationManager = AFHTTPRequestOperationManager()
+        operationManager.requestSerializer = AFJSONRequestSerializer()
+        operationManager.responseSerializer = AFJSONResponseSerializer()
+        
+        operationManager.requestSerializer.setValue("application/json", forHTTPHeaderField: "Accept")
+        operationManager.requestSerializer.setValue("application/json", forHTTPHeaderField: "Content-Type")
+        
+        let params = ["user":
+            [
+                "first_name": firstName,
```

Comment:
hmm.. user UserConstants for this? this would be a good example for when wed need those constants